invenio_rdm_records/resources/serializers/dublincore/schema.py

```diff
@@ -0,0 +1,189 @@
+
+from invenio_access.permissions import system_identity
+from invenio_vocabularies.proxies import current_service as vocabulary_service
+from marshmallow import Schema, fields, missing
+
+
+class DublinCore(Schema):
+    """Schema for Dublin Core in JSON."""
+
+    contributors = fields.Method('get_contributors')
+    titles = fields.Method('get_titles')
+    creators = fields.Method('get_creators')
+    identifiers = fields.Method('get_identifiers')
+    relations = fields.Method('get_relations')
+    rights = fields.Method('get_rights')
+    dates = fields.Method('get_dates')
+    subjects = fields.Method('get_subjects')
+    descriptions = fields.Method('get_descriptions')
+    publishers = fields.Method('get_publishers')
+    types = fields.Method('get_types')
+    sources = fields.Method('get_sources')
+    languages = fields.Method('get_languages')
+    coverage = fields.Method('get_locations')
+    formats = fields.Method('get_formats')
+
+    def get_titles(self, obj):
+        """Get titles."""
+        return [obj['metadata']['title']]
+
+    def get_identifiers(self, obj):
+        """Get identifiers."""
+        items = []
+        items.extend(
+            i['identifier'] for i in obj['metadata'].get('identifiers', [])
+        )
+        items.extend(p['identifier'] for p in obj.get('pids', {}).values())
+
+        return items
+
+    def get_creators(self, obj):
+        """Get creators."""
+        return [
+            c['person_or_org']['name']
+            for c in obj['metadata'].get('creators', [])
+        ]
+
+    def get_relations(self, obj):
+        """Get relations."""
+        rels = []
+
+        # Fundings
+        fundings = obj['metadata'].get('funding', [])
+        for f in fundings:
+            award = f.get('award')
+            if award:
+                rels.append(
+                    u'{0}:{1}'.format(award['scheme'], award['identifier'])
+                )
+
+            funder = f.get('funder', {})
+            if funder:
+                rels.append(
+                    u'{0}:{1}'.format(funder['scheme'], funder['identifier'])
+                )
+
+        # Alternate identifiers
+        for a in obj['metadata'].get('alternate_identifiers', []):
+            rels.append(u'{0}:{1}'.format(a['scheme'], a['identifier']))
+
+        # Related identifiers
+        for a in obj['metadata'].get('related_identifiers', []):
+            rels.append(u'{0}:{1}'.format(a['scheme'], a['identifier']))
+
+        return rels
+
+    def get_rights(self, obj):
+        """Get rights."""
+        rights = []
+        for right in obj['metadata'].get('rights', []):
+            rights.append(right['title'])
+
+            license_url = right.get('link')
+            if license_url:
+                rights.append(license_url)
+
+        return rights
+
+    def get_dates(self, obj):
+        """Get dates."""
+        dates = [obj["metadata"]["publication_date"]]
+
+        if obj['access']['status'] == 'embargoed':
```

Comment:
There's a special encoding for dates embargo dates: https://guidelines.openaire.eu/en/latest/literature/field_embargoenddate.html